cli2/cli.ts

```diff
@@ -181,11 +182,143 @@ function reportFileChangeStats(fileChanges: FileChanges.FileChanges): boolean {
     }
 }
 
+type genOptions ={
+    limitToModules: string[] | null;
+    modulesToInclude: string;
+    targetVersion: string;
+}
+
+const gen = async (input:string, outputPath: string, options:genOptions ) => {
+    await fsMakeDir(outputPath,{
+        recursive:true
+    })
+    const morphirIrJson: Buffer = await fsReadFile(path.resolve(input))
+    const opts = options
+    opts.limitToModules = options.modulesToInclude ? options.modulesToInclude.split(',') : null
+    const fileMap: string[] = await generate(opts, JSON.parse(morphirIrJson.toString()))
+    
+    
+
+    const writePromises = fileMap.map(async ([
+        [dirPath, fileName], content
+    ]:any) => {
+        const fileDir: string = dirPath.reduce((accum: string, next: string)=> path.join(accum,next), outputPath)
+        const filePath: string = path.join(fileDir,fileName)
+        if(await fileExist(filePath)){
+            await fsWriteFile(filePath, content)
+            console.log(`UPDATE - ${filePath}`)
+        }else{
+            await fsMakeDir(fileDir, {
+                recursive: true
+            })
+            await fsWriteFile(filePath, content)
+            console.log(`INSERT - ${filePath}`)
+        }
+    })
+    const filesToDelete = await findFilesToDelete(outputPath, fileMap)
+    const deletePromises =
+        filesToDelete.map(async (fileToDelete: string) => {
+            console.log(`DELETE - ${fileToDelete}`)
+            return fs.unlinkSync(fileToDelete)
+        })
+    copyRedistributables(options, outputPath)
+    return Promise.all(writePromises.concat(deletePromises))
+}
+
+const generate = async (options: genOptions, ir: string): Promise<string[]> => {
+    return new Promise((resolve, reject) =>{
+        worker.ports.jsonDecodeError.subscribe((err: any) =>{
+            reject(err)
+        })
+        worker.ports.generateResult.subscribe(([err, ok]:any) => {
+            if (err) {
+                reject(err)
+            } else {
+                resolve(ok)
+            }
+        })
+
+        worker.ports.generate.send([options, ir])
+    })
+}
+
+const fileExist = async (filePath:string) => {
+    return new Promise((resolve, reject)=>{
+        fs.access(filePath, fs.constants.F_OK, (err) =>{
+            if(err){
+                resolve(false)
+            }else{
```

Comment:
You should pay more attention to code formatting. The use of spaces is not consistent here. There should be spaces around `else` but also around the `=>` above. Best option would be to rely on some code formatting tool.